lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/util/batch/AzureBatchHelper.java

```diff
@@ -52,16 +54,35 @@
 
   private final BatchClient client;
   private final PoolInformation poolInfo;
+  private final TcpPortProvider portProvider;
+  private final ContainerRegistry containerRegistry;
+  private final String containerImageName;
 
   @Inject
   public AzureBatchHelper(
       final AzureBatchFileNames azureBatchFileNames,
       final IAzureBatchCredentialProvider credentialProvider,
+      final TcpPortProvider portProvider,
+      @Parameter(ContainerRegistryServer.class) final String containerRegistryServer,
+      @Parameter(ContainerRegistryUsername.class) final String containerRegistryUsername,
+      @Parameter(ContainerRegistryPassword.class) final String containerRegistryPassword,
+      @Parameter(ContainerImageName.class) final String containerImageName,
```

Comment:
This is a lot of parameters. Can we encapsulate all the container logic into some a ContainerProvider class?

Reply:
Fixed in iteration #9.

---
In reply to: [196513152](https://github.com/sharathmalladi/reef/pull/5#discussion_r196513152) [](ancestors = 196513152)